packages/teleport-uidl-validator/src/validator/utils.ts

```diff
@@ -99,6 +99,17 @@ export const checkDynamicDefinitions = (input: Record<string, unknown>) => {
   const errors: string[] = []
 
   UIDLUtils.traverseNodes(input.node as UIDLNode, (node) => {
+    if (node.type === 'element' && node.content.events) {
+      Object.keys(node.content.events).forEach((eventKey) => {
+        node.content.events[eventKey].forEach((event) => {
+          if (!stateKeys.includes(event.modifies)) {
+            const errorMsg = `"${event.modifies}" is used in events, but not defined in state. Please add it in stateDefinitions`
```

Comment:
Please update the error message to 

```js
`"${event.modifies}" is used in events, but not defined. Please add it in stateDefinitions`
```

Because, we currently support event modifiers only for stateDefinitions.

Reply:
Can we also add an additional check if the `modifies` is defined in `stateDefinitions`.

```js
typeof event.newState === input.stateDefinitions[event.modifies]?.type
```

So, we are sure that the newState is in sync with the `stateDefinition` in UIDL.